### PR TITLE
[AUTO-MERGE] Merge main into develop via merge-main-to-develop

### PR DESCRIPTION
**Pull request by idanov**
A new change in main cannot be merged into develop as part of the regular sync job, hence this PR. Please resolve the conflicts manually, and make sure to obtain 2 approvals once the builds pass.

### IMPORTANT NOTICE

Please let CircleCI merge this PR automatically, with merge commit enabled.

### CODE REVIEW

**Comment by datajoely** [comment]
@lorenabalan can you please validate that `nodes_and_pipelines/micro_packaging.md` and `nodes_and_pipelines/modular_pipelines.md` are up to date for 0.18.0 as discussed in https://github.com/kedro-org/kedro/pull/1128

I don't entirely follow what the action is based on this convo:

image

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+You can pull a modular pipeline from different locations, including local storage, PyPI and the cloud:
+
+```eval_rst
++--------------------------------+--------------------------------------------------------------------------------------+
+| Operation                      | Command                                                                              |
++================================+======================================================================================+
+| Pulling from a local directory | ``kedro pipeline pull <project-root>/src/dist/<pipeline_name>-0.1-py3-none-any.whl`` |
++--------------------------------+--------------------------------------------------------------------------------------+
+| Pull from cloud storage        | ``kedro pipeline pull s3://my_bucket/<pipeline_name>-0.1-py3-none-any.whl``          |
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:84]: These should point at `.tar.gz` files, not `.whl`.

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
++--------------------------------+--------------------------------------------------------------------------------------+
+```
+
+### Providing `fsspec` arguments
+
+* If you are pulling the pipeline from a location that isn't PyPI, Kedro uses [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/) to locate and pull down your pipeline.
+* You can use the `--fs-args` option to point to a YAML that contains the required configuration.
+
+```bash
+kedro pipeline pull https://<url-to-pipeline.whl> --fs-args pipeline_pull_args.yml
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:96]: Same here, no whl.

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+## Pull multiple modular pipelines
+
+* To pull multiple modular pipelines in bulk, run `kedro pipeline pull --all`.
+* This will pull and unpack all pipelines specified in the `tool.kedro.pipeline.pull` manifest section of the project's `pyproject.toml` file:
+
+```toml
+[tool.kedro.pipeline.pull]
+"src/dist/first-pipeline-0.1-py3-none-any.whl" = {}
+"https://www.url.to/second-pipeline.whl" = {alias = "aliased_pipeline", fs-args = "pipeline_pull_args.yml"}
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:114]: No whl. Please see release notes on develop, there's some examples in the migration guide for 0.18.

**Review by lorenabalan — CHANGES_REQUESTED** [review]
Nearly there. The micropackaging file needs a bit more work to bring it in line with the release notes of 0.18.

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+If you plan to publish your packaged modular pipeline to some Python package repository like [PyPI](https://pypi.org/), you need to make sure that your modular pipeline name doesn't clash with any of the existing packages in that repository. However, there is no need to rename any of your source files if that is the case. Simply alias your package with a new name by running `kedro pipeline package --alias <new_package_name> <pipeline_name>`.
+
+In addition to [PyPI](https://pypi.org/), you can also share the packaged wheel file directly, or via a cloud storage such as AWS S3.
+
+## Package multiple modular pipelines
+
+To package multiple modular pipelines in bulk, run `kedro pipeline package --all`. This will package all pipelines specified in the `tool.kedro.pipeline.package` manifest section of the project's `pyproject.toml` file:
+
+```toml
```
Comment (datajoely) [inline docs/source/nodes_and_pipelines/micro_packaging.md:46]: @lorenabalan is this command still valid?

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+## Pull a modular pipeline
+
+You can pull a modular pipeline from a wheel file by executing `kedro pipeline pull <package_name>`.
+
+* The `<package_name>` must either be a package name on PyPI or a path to the wheel file.
+* Kedro will unpack the wheel file, and install the files in following locations in your Kedro project:
+  * All the modular pipeline code in `src/<python_package>/pipelines/<pipeline_name>/`
+  * Configuration files in `conf/<env>/parameters/<pipeline_name>.yml`, where `<env>` defaults to `base`.
+  * To place parameters from a different config environment, run `kedro pipeline pull <pipeline_name> --env <env_name>`
+  * Pipeline unit tests in `src/tests/pipelines/<pipeline_name>`
```
[the remark this answers is not in the document]
Comment (datajoely) [reply docs/source/nodes_and_pipelines/micro_packaging.md:68]: Done

File: docs/source/nodes_and_pipelines/micro_packaging.md (hunk 1 above)
Comment (datajoely) [reply docs/source/nodes_and_pipelines/micro_packaging.md:84]: Done

File: docs/source/nodes_and_pipelines/micro_packaging.md (hunk 2 above)
Comment (datajoely) [reply docs/source/nodes_and_pipelines/micro_packaging.md:96]: Done

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+# Micro-packaging
+
+Micro-packaging allows users to share Kedro pipelines across codebases, organisations and beyond.
+
+## Package a modular pipeline
+
+You can package a modular pipeline by executing: `kedro pipeline package <pipeline_name>`
+
+* This will generate a new [wheel file](https://pythonwheels.com/) for this pipeline.
+* By default, the wheel file will be saved into `src/dist` directory inside your project.
```
[the remark this answers is not in the document]
Comment (datajoely) [reply docs/source/nodes_and_pipelines/micro_packaging.md:11]: Done

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
+
+# Micro-packaging
+
+Micro-packaging allows users to share Kedro pipelines across codebases, organisations and beyond.
+
+## Package a modular pipeline
+
+You can package a modular pipeline by executing: `kedro pipeline package <pipeline_name>`
+
+* This will generate a new [wheel file](https://pythonwheels.com/) for this pipeline.
```
[the remark this answers is not in the document]
Comment (datajoely) [reply docs/source/nodes_and_pipelines/micro_packaging.md:10]: Done

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
+
+# Micro-packaging
+
+Micro-packaging allows users to share Kedro pipelines across codebases, organisations and beyond.
+
+## Package a modular pipeline
+
+You can package a modular pipeline by executing: `kedro pipeline package <pipeline_name>`
```
[the remark this answers is not in the document]
Comment (datajoely) [reply docs/source/nodes_and_pipelines/micro_packaging.md:8]: Done

File: docs/source/development/commands_reference.md [hunk 9]
```diff
@@ -300,7 +300,7 @@ The following command packages all the files related to a modular pipeline into
 kedro pipeline package <pipeline_module_path>
 ```
 
-Further information is available in the [pipeline documentation](../nodes_and_pipelines/modular_pipelines.md#package-a-modular-pipeline).
+Further information is available in the [pipeline documentation](../nodes_and_pipelines/modular_pipelines.md).
```
[the remark this answers is not in the document]
Comment (datajoely) [reply docs/source/development/commands_reference.md:303]: Done

**Comment by datajoely** [comment]
@lorenabalan I think I've addressed all comments - can you verify if `micro-packaging.md` meets your understanding?

File: docs/source/nodes_and_pipelines/micro_packaging.md (hunk 4 above)
Comment (lorenabalan) [reply docs/source/nodes_and_pipelines/micro_packaging.md:47]: The command is valid but the explanation is still too modular-pipeline focused. `first_package` and `second_package` are names of python packages in `src/<package_name>`, not necessarily modular pipeline folders. If they want modular pipeline folders, they would have to be specific, i.e. `pipelines.first_package`, `pipelines.first_package`.

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+* The keys (`first_package`, `second_pacakge`) are the names of the modular pipeline folders within the codebase.
+* The values are the options accepted by the `kedro pipeline package <pipeline_name>` CLI command.
+
+```eval_rst
+.. note::  Make sure `destination` is specified as a POSIX path even when working on a Windows machine.
+```
+
+```eval_rst
+.. note::  The examples above apply to any generic Python package, modular pipelines fall under this category and can be easily addressed via the ``piplines.pipeline_name`` syntax.
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:61]:
```suggestion
.. note::  The examples above apply to any generic Python package, modular pipelines fall under this category and can be easily addressed via the ``pipelines.pipeline_name`` syntax.
```

File: docs/source/nodes_and_pipelines/micro_packaging.md (hunk 4 above)
Comment (lorenabalan) [reply docs/source/nodes_and_pipelines/micro_packaging.md:47]: Actually I see you sort of addressed this in the `note` below, but I'm wondering if that's visible enough. Happy to approve as is, and if you have other ideas for rewording it then great, if not that's fine.

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+```
+
+
+## Pull a modular pipeline
+
+You can pull a modular pipeline from a wheel file by executing `kedro pipeline pull <package_name>`.
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:67]:
```suggestion
You can pull a modular pipeline from a tar file by executing `kedro pipeline pull <package_name>`.
```

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
+* The `<package_name>` must either be a package name on PyPI or a path to the wheel file.
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:69]:
```suggestion
* The `<package_name>` must either be a package name on PyPI or a path to the tar file.
```

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+* To pull multiple modular pipelines in bulk, run `kedro pipeline pull --all`.
+* This will pull and unpack all pipelines specified in the `tool.kedro.pipeline.pull` manifest section of the project's `pyproject.toml` file:
+
+```toml
+[tool.kedro.pipeline.pull]
+"src/dist/first-pipeline-0.1-py3-none-any.tar.gz" = {}
+"https://www.url.to/second-pipeline.tar.gz" = {alias = "aliased_pipeline", fs-args = "pipeline_pull_args.yml"}
+```
+
+* The keys (wheel references in this case) are the package paths
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:123]:
```suggestion
* The keys (tar references in this case) are the package paths
```

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+
+Micro-packaging allows users to share Kedro pipelines across codebases, organisations and beyond.
+
+## Package a modular pipeline
+
+You can package a modular pipeline by executing: `kedro pipeline package pipelines.<pipeline_name>`
+
+
+* This will generate a new [tar](https://docs.python.org/3/distutils/sourcedist.html) file for this pipeline.
+* By default, the wheel file will be saved into `dist` directory inside your project.
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:12]:
```suggestion
* By default, the tar file will be saved into `dist` directory inside your project.
```

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+    │   ├── __init__.py
+    │   └── pipelines
+    │       └── {{pipeline_name}}    <-- Pipeline folder
+    └── tests
+        ├── __init__.py
+        └── pipelines
+            └── {{pipeline_name}}    <-- Pipeline tests
+```
+
+Kedro will also include any requirements found in `src/<python_package>/pipelines/<pipeline_name>/requirements.txt` in the modular pipeline wheel file. These requirements will later be taken into account when pulling a pipeline via `kedro pipeline pull`.
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:33]:
```suggestion
Kedro will also include any requirements found in `src/<python_package>/pipelines/<pipeline_name>/requirements.txt` in the modular pipeline tar file. These requirements will later be taken into account when pulling a pipeline via `kedro pipeline pull`.
```

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+
+```eval_rst
+.. note::  Kedro will not package the catalog config files even if those are present in ``conf/<env>/catalog/<pipeline_name>.yml``.
+```
+
+If you plan to publish your packaged modular pipeline to some Python package repository like [PyPI](https://pypi.org/), you need to make sure that your modular pipeline name doesn't clash with any of the existing packages in that repository. However, there is no need to rename any of your source files if that is the case. Simply alias your package with a new name by running `kedro pipeline package --alias <new_package_name> <pipeline_name>`.
+
+In addition to [PyPI](https://pypi.org/), you can also share the packaged wheel file directly, or via a cloud storage such as AWS S3.
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:41]:
```suggestion
In addition to [PyPI](https://pypi.org/), you can also share the packaged tar file directly, or via a cloud storage such as AWS S3.
```

File: docs/source/nodes_and_pipelines/micro_packaging.md [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+.. note::  The examples above apply to any generic Python package, modular pipelines fall under this category and can be easily addressed via the ``piplines.pipeline_name`` syntax.
+```
+
+
+## Pull a modular pipeline
+
+You can pull a modular pipeline from a wheel file by executing `kedro pipeline pull <package_name>`.
+
+* The `<package_name>` must either be a package name on PyPI or a path to the wheel file.
+* Kedro will unpack the wheel file, and install the files in following locations in your Kedro project:
```
Comment (lorenabalan) [inline docs/source/nodes_and_pipelines/micro_packaging.md:70]:
```suggestion
* Kedro will unpack the tar file, and install the files in following locations in your Kedro project:
```

**Review by lorenabalan — APPROVED** [review]
I would double check there are no leftover `wheel` references on the page, but otherwise I think this is fine. Thank you resolving the conflicts!

**Review by datajoely — APPROVED** [review]

**Pull request merged**